evm/db/state.py

```diff
@@ -300,7 +300,8 @@ def account_has_code_or_nonce(self, address):
         return self.get_code(address) != b'' or self.get_nonce(address) != 0
 
     def touch_account(self, address):
-        if not self.account_exists(address):
+        do_not_touch = self.account_has_code_or_nonce(address) or self.get_balance(address) != 0
```

Comment:
can we invert this.

```python
if self.account_has_code_or_nonce(address):
    should_touch = False
elif self.get_balance(address) != 0:
    should_touch = False
else:
    should_touch = True:

if should_touch:
    ...
```

It's longer and more explicit, but I think it's easier to follow the logic.
should_touch = 
    not self.account_has_
)